src/parser/warrior/fury/modules/buffdebuff/Enrage.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+
+import Analyzer from 'parser/core/Analyzer';
+
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import Events from 'parser/core/Events';
+import { SELECTED_PLAYER } from 'parser/core/EventFilter';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage, formatNumber, formatThousands } from 'common/format';
+import StatisticBox from 'interface/others/StatisticBox';
+import calculateEffectiveDamage from 'parser/core/calculateEffectiveDamage';
+
+class Enrage extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+    abilityTracker: AbilityTracker,
+  }
+
+  totalDamage = 0;
+  damage = 0;
+
+  constructor(...args) {
+    super(...args);
+
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onPlayerDamage);
+  }
+
+  onPlayerDamage(event) {
+    if (this.selectedCombatant.hasBuff(SPELLS.ENRAGE.id)) {
+      this.damage += calculateEffectiveDamage(event, this.statTracker.currentMasteryPercentage);
+      this.totalDamage += event.amount;
+    }
+  }
+
+  get uptime() {
+    return this.selectedCombatant.getBuffUptime(SPELLS.ENRAGE.id) / this.owner.fightDuration;
+  }
+
+  get damageTotal () {
+    return this.damage / (this.owner.fightDuration / 1000);
+  }
+
+  get damageTotalPercent() {
+    return this.owner.getPercentageOfTotalDamageDone(this.damage);
+  }
+
+  get DPSPercent() {
+    return formatPercentage(this.owner.getPercentageOfTotalDamageDone(this.damage) / this.owner.getPercentageOfTotalDamageDone(this.totalDamage));
+  }
+
+  get suggestionThresholds() {
+    return {
+      isLessThan: {
+        minor: 0.7,
+        average: 0.65,
+        major: 0.6,
+      },
+      style: 'percentage',
+    };
+  }
+
+  suggestions(when) {
+    const {
```

Comment:
I dont think you need this part here, all that gets done automatically if you do the below

suggestions(when) {
    when(this.suggestionThresholds)
        .addSuggestion((suggest, actual, recommended) => {

Reply:
works only if you add `actual: this.enrageUptime, ` inside the suggestion thresholds tho

Reply:
yea i missed that. Here is an example from one of the mage modules

  get utilSuggestionThresholds() {
    return {
      actual: this.utilPercent,
      isLessThan: {
        minor: 0.95,
        average: 0.90,
        major: 0.80,
      },
      style: 'percentage',
    };
  }

Reply:
basically change this

```
suggestions(when) {
    const {
      isLessThan: {
        minor,
        average,
        major,
      },
    } = this.suggestionThresholds;
     when(this.enrageUptime).isLessThan(minor)
``

to this
```
suggestions(when) {
    when(this.suggestionThresholds)
```

and add `actual: this.enrageUptime,` above `isLessThan: {` in the suggestionThresholds